lib/mix/lib/mix.ex

```diff
@@ -199,8 +199,20 @@ defmodule Mix do
     Supervisor.start_link(children, opts)
   end
 
-  @doc """
+  @doc ~S"""
   Returns the Mix environment.
+
+  This function should not be used at runtime. Mix is a build tool and may not
+  be available after the code is compiled (for example in a release).
+
+  To differentiate the program behavior depending on the environment, it is
+  recommended to use application environment through `Application.get_env/3`.
+  Proper configuration can be set in the `config/config.exs` file or a
+  per-environment config file (for example `config/test.exs`) after ensuring
+  the main `config/config.exs` file contains a proper import instruction:
+
+      import_config "#{Mix.env}.exs"
```

Comment:
I think this is too much information because most of it is duplicated (from `import_config/1` for example). I would say something like:

> This function should not be used at runtime. Mix is a build tool and may not be available after the code is compiled (for example in a release).
>
> To differentiate runtime behaviour depending on the environment, it's recommended to use application environment (see `Application.get_env/3`). Proper configuration can be set in `Mix.Config` files, often per-environment (see `Mix.Config.import_config/1` for more information).

Wdyt?